src/parachain/issue.ts

```diff
@@ -157,16 +159,10 @@ export class DefaultIssueAPI extends DefaultTransactionAPI implements IssueAPI
 
     }
 
-    async execute(requestId: string, btcTxId: string, merkleProof?: Bytes, rawTx?: Bytes): Promise<void> {
+    async execute(requestId: string, btcTxId?: string, merkleProof?: Bytes, rawTx?: Bytes): Promise<void> {
         const parsedRequestId = this.api.createType("H256", "0x" + requestId);
-        const parsedBtcTxId = this.api.createType(
-            "H256",
-            "0x" + Buffer.from(btcTxId, "hex").reverse().toString("hex")
-        );
-        if (!merkleProof || !rawTx) {
-            [merkleProof, rawTx] = await this.electrsAPI.getParsedExecutionParameters(btcTxId);
-        }
-        const executeIssueTx = this.api.tx.issue.executeIssue(parsedRequestId, parsedBtcTxId, merkleProof, rawTx);
+        [merkleProof, rawTx] = await getTxProof(this.electrsAPI, btcTxId, merkleProof, rawTx);
```

Comment:
In case a user does not submit the btcTxId, I'd assume you need to fetch this here? Also this call can be optional in case the user submits the proof and rawTX